deploy/virtualbox_xenial_gui_creation/Vagrantfile

```diff
@@ -1,67 +1,205 @@
 
-  config.vm.provider "virtualbox" do |vb|
-    vb.gui = true
-    vb.memory = "2048"
+  config.vm.provider "virtualbox" do |v|
+    v.name="vag-u1604-xf-creation"
+    v.memory = 4096
+    v.cpus = 4
+    v.gui = true
   end
 
   config.vm.provision "ubuntu-setup", type: "shell", privileged: false, inline: <<~SHELL
-    set -eux
+    set -eux -o pipefail
+
+    function tryharderto() { i=0; until "$@"; do ((++i > 3)) && false; echo >&2 Retrying $i; sleep 2m; done }
 
     # Disable automatic updates
     sudo sed -i 's/APT::Periodic::Update-Package-Lists "1";/APT::Periodic::Update-Package-Lists "0";/g' /etc/apt/apt.conf.d/10periodic
     sudo apt-get purge -y unattended-upgrades
 
-    CODENAME="$(lsb_release -cs)"
-    MIRRORS="deb mirror://mirrors.ubuntu.com/mirrors.txt $CODENAME main restricted universe multiverse
-    deb mirror://mirrors.ubuntu.com/mirrors.txt $CODENAME-updates main restricted universe multiverse
-    deb mirror://mirrors.ubuntu.com/mirrors.txt $CODENAME-backports main restricted universe multiverse
-    deb mirror://mirrors.ubuntu.com/mirrors.txt $CODENAME-security main restricted universe multiverse
-    "
-
-    # Add sources to the top of sources.list if they're not already there
-    grep --fixed-strings --silent 'deb mirror://mirrors.ubuntu.com/mirrors.txt' /etc/apt/sources.list || { { echo "$MIRRORS" && cat /etc/apt/sources.list; } > sources.list.tmp && sudo mv sources.list.tmp /etc/apt/sources.list; }
+    # Pre-answer grub-pc dpkg dialog
+    echo grub-pc grub-pc/install_devices multiselect $(ls /dev/disk/by-id/ata-VBOX_HARDDISK_VB*|grep -v part) | sudo debconf-set-selections
+    echo grub-pc grub-pc/install_devices_disks_changed multiselect $(ls /dev/disk/by-id/ata-VBOX_HARDDISK_VB*|grep -v part) | sudo debconf-set-selections
 
-    sudo apt-get update
-    sudo apt-get -y upgrade
+    tryharderto sudo apt-get update
+    tryharderto sudo apt-get -y upgrade
 
     # In order to run gsettings over SSH we have to set DBUS_SESSION_BUS_ADDRESS. See https://askubuntu.com/a/457023
     export DBUS_SESSION_BUS_ADDRESS="$(sudo cat /proc/$(pidof -s pulseaudio)/environ | tr '\\0' '\\n' | grep --text DBUS_SESSION_BUS_ADDRESS | cut -d '=' -f 2-)"
-    if [ "$DBUS_SESSION_BUS_ADDRESS" != "" ]; then
-      gsettings set org.gnome.desktop.screensaver lock-enabled false
-      gsettings set org.gnome.desktop.session idle-delay 0
-      gsettings set com.canonical.Unity.Lenses remote-content-search none
-      gsettings set com.canonical.Unity.Launcher favorites "['application://org.gnome.Nautilus.desktop', 'application://firefox.desktop', 'application://unity-control-center.desktop', 'unity://running-apps', 'application://gnome-terminal.desktop', 'unity://expo-icon', 'unity://devices']"
-    else
-      echo ***COULD NOT MODIFY GSETTINGS***
-    fi
+    [[ -n "$DBUS_SESSION_BUS_ADDRESS" ]] || exit 1
+    # Somehow dconf was owned by root.
+    sudo chown -R vagrant:vagrant /run/user/1000/dconf
```

Comment:
I think it failed here with this error:
```
    default: +++ sudo cat /proc/1227/environ
    default: ++ export DBUS_SESSION_BUS_ADDRESS=unix:abstract=/tmp/dbus-FIRX4dFQoA,guid=3f0788233baf403c328f7aaf5ba6332f
    default: ++ DBUS_SESSION_BUS_ADDRESS=unix:abstract=/tmp/dbus-FIRX4dFQoA,guid=3f0788233baf403c328f7aaf5ba6332f
    default: ++ [[ -n unix:abstract=/tmp/dbus-FIRX4dFQoA,guid=3f0788233baf403c328f7aaf5ba6332f ]]
    default: ++ sudo chown -R vagrant:vagrant /run/user/1000/dconf
    default: chown: cannot access '/run/user/1000/dconf'
    default: : No such file or directory
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

Reply:
This is the error you get if you don't log into the machine while provisioning is running. An improvement might be to have the provisioning script pause and prompt the user to login if it gets to this point and finds the directory missing.

Reply:
Yes I did see that instruction and had logged in previously but must have forgotten on the one that actually failed. Thanx